### PR TITLE
tested out navigation highlighting

### PR DESCRIPTION
**Pull request by sherwoodf**
The Figma nav bars look like this:

image

I thought i would try to recreate those - this is a start but warrants further testing to make sure it works.

Additional things this brought up:

### Path/folder structure
Currently only highlights the nav bar if it is directly a page that you click on.

One way to make it easier in the code to work out whether you're on a 'help' page is to have all the 'Help' pages could be under a 'help' folder/path. I think the folder would be helpful for organising pages anyway.  However this would break a lot of links.

### CSS imports
I had to remove some of the classes used by css to get margins working so that the line lines up.

I'm not too familiar with the style sheets we're importing - do we want to rely on these external sheets, or take more control of them in the future?

### Left bar navigation
Currently not done anything to highlight the left bar navigation. Additionally, the Submit page doesn't have any left nav bar and so looks a bit odd compared to the rest of the pages.

### NOTE doesn't work on preview...

Should look like:

image

### CODE REVIEW

File: astro.config.mjs [hunk 1]
```diff
@@ -13,5 +13,28 @@ export default defineConfig({
   vite: {
     plugins: [yaml()]
   },
-  base: "/bioimage-archive"
+  base: "/bioimage-archive",
+  redirects: {
```
Comment (sherwoodf) [inline astro.config.mjs:17]: i decided to be exhaustive and list all redirects i caused - but many of these pages dont really exist on the current website, so i don't know that we need to be worried about them.

File: src/pages/help.mdx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+title: Help Overview
+layout: ../layouts/HelpLayout.astro
+---
+import onlineImage from '/src/assets/bioimage-archive/online_tutorial_image.png';
+
+## Help Overview
+
+Please use the sidebar menu on the left to navigate the help pages. 
+
+If this is the first time you are submitting to the BioImage Archive, we have a comprehensive online tutorial that covers all you need to know to get started:
```
Comment (sherwoodf) [inline src/pages/help.mdx:11]: i didn't think we had enough/a obvious link to our tutorial, so added one here.

File: README.md [hunk 3]
```diff
@@ -10,5 +10,23 @@ All commands are run from the root of the project, from a terminal:
 | `npm run dev`             | Starts local dev server at `localhost:4321`      |
 | `npm run build`           | Build your production site to `./dist/`          |
 | `npm run preview`         | Preview your build locally, before deploying     |
+| `npm run local`           | Equivalent to build && preview                   |
 | `npm run astro ...`       | Run CLI commands like `astro add`, `astro check` |
 | `npm run astro -- --help` | Get help using the Astro CLI                     |
+
+Note that:
+`npm run dev` will update as you change things but may give a false impression of the final result, whereas running `npm run build` and then `npm run preview` will recreate what netlify runs, and will provide an accurate display of the website.
```
Comment (AybukeKY) [inline README.md:18]: This is a great functionality, thank you. It works fine for me.

File: src/components/Navigation.astro [hunk 4]
```diff
@@ -9,14 +9,30 @@ const pageData = {
     policies: "Policies",
     "about-us": "About"
 }
+const pathname = Astro.url.pathname;
+const basePath = pathname.split('/')[2];
```
Comment (AybukeKY) [inline src/components/Navigation.astro:13]: This doesn't look very future-proof. Is there a different way to do this?

File: src/pages/project-developments.mdx
Comment (AybukeKY) [inline src/pages/project-developments.mdx:1]: The two top images (development-milestones.png and bioimaging_governance_phase2.png) do not work. The links do either need to be relative, e.g. ../assets/biomage-archive/development-milestones.png or need to be imported and linked as .

**Review by AybukeKY — COMMENTED** [review]
I have left a few comments, for two possible minor changes. Otherwise looks great and work when tested.

File: README.md [hunk 5]
```diff
@@ -10,5 +10,23 @@ All commands are run from the root of the project, from a terminal:
...
+Prerequisites
+* install linkchecker: https://linkchecker.github.io/linkchecker/install.html
+
+To test:
+* build and preview locally: `npm run local`
+* run: `linkchecker http://localhost:4321/bioimage-archive/ -o failures --no-warnings --ignore-url '.*\/vis\/?.*' --ignore-url '.*\/dataset\/?.*' --ignore-url '.*\/sodataset\/?.*' --ignore-url '.*\/aidataset\/?.*'` *
+
+\* Note that on mac you may have to modify permissions of the /Users/$USER/.config/linkchecker to get this command to work.
+
+You can also run this on the files genated by the build by runnign `npm run build` and then `linkchecker dist/bioimage-archive/ ...`
```
[the remark this answers is not in the document]
Comment (tzulueta) [reply README.md]: Got the exact same errors as @AybukeKY

[the remark this answers is not in the document]
Comment (sherwoodf) [reply README.md:32]: To be clear i meant running:

```
linkchecker dist/bioimage-archive/ -o failures --no-warnings --ignore-url '.*\/vis\/?.*' --ignore-url '.*\/dataset\/?.*' --ignore-url '.*\/sodataset\/?.*' --ignore-url '.*\/aidataset\/?.*'
``` 
Which gives 5 broken links that are all to do with a weird image page (imlist/index.html) that a user cannot access without knowing the URL to begin with (and i've not touched i think)

I think they are unrelated to my changes here. There are links to an access_data_help.html but i didn't touch that page & it doesn't exist, so i don't think the link ever worked.

File: src/components/Navigation.astro (hunk 4 above)
Comment (sherwoodf) [reply src/components/Navigation.astro:13]: I think it depends on how we want to manage this. If we want to make pages handle their own navigational highlighting we would have to wire it through a bunch of pages, which might be messy. If we want to handle it here, we're essentially going to have to do _something_ that is doing a url comparison - which if we ever want to make changes to the base urls will require updates here - but we would need to do that anyway since the navigation is controlled by the pageData list above.

In short, i think this is basically as future proof as having the map in this file listing the navigation options. The front part of the path is at least consistent when generating astro pages on different environments - i had to deal with optional trailing / for the other navigation options that was massive pain.

**Review by tzulueta — COMMENTED** [review]
It looks great and what I tested worked fine, so I have nothing more to add. 

Not for this PR, but one thought about what @sherwoodf mentioned about the Submit page missing the left bar navigation. Could it be useful to move the the submit-related help pages to the submit page? I think it could be helpful to at least have the annotations submission info in there.

File: src/components/Navigation.astro (hunk 4 above)
Comment (AybukeKY) [reply src/components/Navigation.astro:13]: Fair enough. I don't think this is massively different than individual pages having local path links to other components or pages.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (AybukeKY) [reply README.md]: Ah yes I checked a few previous versions and they don't have access_data_help either. I must have forgotten to migrate it. Will add a ticket.

**Review by AybukeKY — APPROVED** [review]

**Comment by sherwoodf** [comment]
Thanks for a review - i will merge this but either make updates immediately in the next branch i'm working on, or make some tickets to keep track of, the changes to the pages you've suggest.
